### ISSUE TITLE
Feature Request: PCI passthru

### ISSUE THREAD

**Issue by misterfinster** [churchers/vm-bhyve#38]
Would also be extra handy to support pci pass thru (including -S parameter for bhyveload/bhyve for wiring memory)

**Comment by WanpengQian** [comment]
From the config.sample:

```
# passthru0
# Add a pass-through PCI device to the virtual machine. This allows the guest
# to access a hardware device no differently than if it was running on bare
# metal. The value of this option is the B/S/F of the appropriate device.
# e.g "3/0/0"
# 
# Please note that in order to stop the bhyve host from attaching to the device,
# there are some steps required to reserve the device in /boot/loader.conf.
# 
# More details can be found in the FreeBSD bhyve wiki pages
# 
passthru0=""
```

you can add passthru0 to vm config. and vm-bhyve will add -S to command line when starting vm.
but I didn't get it work. I don't know something wrong with my hardware or bhyve is broken in CURRENT.

**Comment by misterfinster** [comment]
bhyve on freebsd-current works with -S and pcipassthru with my setup.  I've scanned through the vm-bhyve scripts and there isn't any -S added in the bhyveload/grub-bhyve and in the bhyve call to wire the memory.  This prevents vm-bhyve from supporting pass thru.  I suppose this is a script bug?  If I change vm-guest and vm-run to force all my VM's to add -S to both calls (thus wiring all their memory), passthru works with the vm-byhve scripts.

**Comment by WanpengQian** [comment]
Yes, I check the code. grub-bhyve lacks of a -S option. maybe Matt will have an update later.
and for bhyve, it already has a -S option when passthru device presents in config file.

**Comment by churchers** [comment]
Should be fixed in 0.9.20

**Comment by mlavkin** [comment]
Does it fixed?
